editor/src/components/inspector/sections/component-section/variables-in-scope-utils.ts

```diff
@@ -443,49 +437,16 @@ const filterObjectPropFromVariablesInScope =
     return next
   }
 
-const keepLocalestScope =
-  () =>
-  (variablesInScope: VariableData): VariableData => {
-    let deepestInsertionCeiling = -Infinity
-    Object.values(variablesInScope).forEach((variable) => {
-      if (variable.insertionCeiling.type === 'file-root') {
-        return
-      }
-
-      deepestInsertionCeiling = Math.max(
-        deepestInsertionCeiling,
-        EP.fullDepth(variable.insertionCeiling),
-      )
-    })
-
-    const result: VariableData = {}
-    Object.entries(variablesInScope).forEach(([key, variable]) => {
-      if (variable.insertionCeiling.type === 'file-root') {
-        result[key] = variable
-        return
-      }
-
-      if (EP.fullDepth(variable.insertionCeiling) === deepestInsertionCeiling) {
-        result[key] = variable
-      }
-    })
-
-    return result
-  }
-
 export function useVariablesInScopeForSelectedElement(
   elementPath: ElementPath | null,
-  propertyPath: PropertyPath | null,
+  matchFn: (_: Array<VariableInfo>) => Array<VariableInfo>,
```

Comment:
could you make this a named type that you also use in the matchers? that way a right click -> Find All Implementations can quickly take us to all matchers, no need to jump around in the codebase

Reply:
good idea, pushed it in https://github.com/concrete-utopia/utopia/pull/5987/commits/4abb909eba9f1ebe5ab863a2d5ac217cba897ef8